gef-ui/templates/partials/application-activity.njk

```diff
@@ -21,54 +22,47 @@
 
   <h1 class="govuk-heading-xl" data-cy="main-activity-heading">Activity and comments</h1>
 
-    {% set portalActivitiesWithHTML = [] %}
+  {# 
+    NOTE: we need to use MOJ timeline HTML rather than the nunjucks component.
+    This is because we have conditional custom HTML for different timeline items/activities
+    See the "activity-html" / activityHtml component for examples.
+    The alternative to this is to have complex array mapping, object creation and creation.
+    inside this nunjucks template, which we found to be very hard to read and maintain.
+  #}
```

Comment:
I think it might be possible to use the nunjucks component & abstract the activity html away
e.g. 

```
  {% set portalActivitiesWithHTML = [] %}

  {% for activity in portalActivities %}

    {% set activityToAdd = {
        label: activity.label,
        text: activity.text,
        datetime: activity.datetime,
        byline: activity.byline,
        html: activityHtml.render(activity)
      }
    %}

    {% set portalActivitiesWithHTML = (portalActivitiesWithHTML.push(activityToAdd), portalActivitiesWithHTML) %}
  {% endfor %}

  {{ mojTimeline({
      items: portalActivitiesWithHTML,
      attributes: {
        'data-cy': 'portal-activities-timeline'
      }
    }) }}
```

Reply:
Pretty sure I tried this or similar and it wouldn't work because of the MOJ component and `html: activityHtml.render(activity)`. I.e it would not actually render the HTML.

Will double check.

Reply:
Had a think about this @MarRobSoftwire - even if the suggestion above works, I don't think we should do it.

Really not a fan of constructing and pushing objects in a nunjucks file, same with arrays. I think it's much cleaner to have it simply consuming data from the controller.

Reply:
Happy to leave this then, & see what you mean, though I think this does bring with it a burden of maintaining the component if the timeline is ever updated

Reply:
I do agree, I don't believe there has been an update to the timeline component - if there is, we'll at least only have to manually update it in one place.

I think there's more pros than cons to _not_ using the component here.